src/main/java/org/embulk/input/marketo/rest/MarketoRestClient.java

```diff
@@ -0,0 +1,335 @@
+            throw new DataException(String.format("Can't start job for export Job id : %s, error code: %s, error message: %s", exportId, error.getCode(), error.getMessage()));
+        }
+    }
+
+    /**
+     * Wait for lead bulk extract job
+     * Will block and wait until job status switch to complete
+     * If job run logger than bulk job timeout then will stop and throw exception
+     * If job status is failed or cancel will also throw exception
+     *
+     * @param exportId
+     * @throws InterruptedException
+     */
+    public void waitLeadExportJobComplete(String exportId, int pollingInterval, int waitTimeout) throws InterruptedException
+    {
+        waitExportJobComplete(MarketoRESTEndpoint.GET_LEAD_EXPORT_STATUS, exportId, pollingInterval, waitTimeout);
+    }
+
+    /**
+     * Wait for activites bulk extract job
+     * Will block and wait until job status switch to complete
+     * If job run logger than bulk job timeout then will stop and throw exception
+     * If job status is failed or cancel will also throw exception
+     *
+     * @param exportId
+     * @throws InterruptedException
+     */
+    public void waitActitvityExportJobComplete(String exportId, int pollingInterval, int waitTimeout) throws InterruptedException
+    {
+        waitExportJobComplete(MarketoRESTEndpoint.GET_ACTIVITY_EXPORT_STATUS, exportId, pollingInterval, waitTimeout);
+    }
+
+    private void waitExportJobComplete(MarketoRESTEndpoint marketoRESTEndpoint, String exportId, int pollingInterval, int waitTimeout) throws InterruptedException
+    {
+        long waitTime = 0;
+        while (true) {
+            MarketoResponse<ObjectNode> marketoResponse = doGet(this.endPoint + marketoRESTEndpoint.getEndpoint(
+                    new ImmutableMap.Builder<String, String>().put("export_id", exportId).build()), null, null, new MarketoResponseJetty92EntityReader<ObjectNode>(READ_TIMEOUT_MILLIS));
+            if (marketoResponse.isSuccess()) {
+                ObjectNode objectNode = marketoResponse.getResult().get(0);
+                String status = objectNode.get("status").asText();
+                if (status == null) {
+                    throw new DataException("Can't get bulk extract status export job id: " + exportId);
+                }
+                LOGGER.info("Jobs [{}] status is [{}]", exportId, status);
+                switch (status) {
+                    case "Completed":
+                        return;
+                    case "Failed":
+                        throw new DataException("Bulk extract job failed exportId: " + exportId + " errorMessage: " + objectNode.get("errorMsg").asText());
+                    case "Cancel":
+                        throw new DataException("Bulk extract job canceled, exportId: " + exportId);
+                }
+            }
+            Thread.sleep(pollingInterval * 1000);
+            waitTime = waitTime + pollingInterval;
```

Comment:
If `waitTimeout` is what I'm understanding, i.e. the maximum time to wait for the request to be returned, this code doesn't do as expected. It doesn't count the time spent on doing real request, wait & process the response. It only accumulates the `pollingInterval` which is sleeping time between requests.

Reply:
Fixed